test_pipeline.py

```diff
@@ -0,0 +1,37 @@
+import dlt
+
+from dlt import annotations as a
+from dlt.common import json
+from typing_extensions import Annotated, Never, Optional
+
+
+class Items:
+
+    # metadata for the table, currently not picked up by the pipeline
+    __table__: Annotated[Never, a.TableName("my_items"), a.WriteDisposition("merge")]
```

Comment:
hmmm there must be a better way! ie.
load_info = p.run(data, columns=Annotated[Items, a.TableName("my_items"), a.WriteDisposition("merge")] , table_name="blah")

if items are not annotated then we go to default: Items as name, append and write dispositions

Reply:
also using `columns` for the above is a kind of abuse... let's sync on table level hints. they may require to change our core library. ie. to make resource a generic that takes model as T. also see: https://github.com/dlt-hub/dlt/issues/753

Reply:
or we add "model" to resource definition. but then we are going into a big overhaul of our schema system where relational and python schemas are different.

Reply:
i changed it to accept annotated tables now, so this would work.